salt/grains/core.py

```diff
@@ -2434,6 +2548,52 @@ def _hw_data(osdata):
                 grains['product'] = res.group(1).strip().replace("'", "")
                 break
 
+    elif osdata['kernel'] == 'AIX':
+        data = ""
+        cmd = salt.utils.path.which('prtconf')
+        if cmd:
+            data += __salt__['cmd.run']('{0}'.format(cmd)) + os.linesep
+
```

Comment:
Again `+=`. This is not needed here, especially that `data` is not used after `if cmd` returns `False`.